File: 02 백준/10 문자열/Python/Prob20291.py

"""
title  : 20291. 파일 정리 (Silver 3)
time   : 204ms        324ms
memory : 41104KB      125384KB
"""

import sys

def solution() -> dict:

    input = sys.stdin.readline
    N = int(input())    # 파일 개수

    Extension = {}

    for _ in range(N):
        extension = input().rstrip().split(".")[1]
        Extension[extension] = Extension.get(extension, 0) + 1

    return Extension

answer = solution()
for key in sorted(answer.keys()):
    print(key, answer[key], sep=" ")

File: 02 백준/10 문자열/Java/src/Prob20291.java

/**
* title  : 20291. 파일 정리 (Silver 3)
* time   : 784ms
* memory : 65872KB
*/

import java.io.*;
import java.util.*;

public class Prob20291 {

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        int N = Integer.parseInt(br.readLine());

        HashMap<String, Integer> nameMap = new HashMap<>();

        for (int i = 0; i < N; i++) {
            String name = (br.readLine()).split("\\.")[1];
            nameMap.put(name, nameMap.getOrDefault(name, 0) + 1);
        }

        List<String> keySet = new ArrayList<>(nameMap.keySet());
        Collections.sort(keySet);

        for (String key : keySet) {
            sb.append(key).append(" ").append(nameMap.get(key)).append("\n");
        }

        System.out.println(sb);
    }
}